### ISSUE TITLE
Adding a getter for the WiFiClientSecure object

### ISSUE THREAD

**Issue by Duckle29** [maakbaas/esp8266-iot-framework#75]
Hey there. I'm looking at making some MQTT based IoT nodes, and really really really want to use an external MQTT service. Because of this I really want to use SSL, and this framework is so far the only proper SSL implementation for ESP8266s

I was wondering if it'd be feasible to add a getter for the `WiFiClientSecure` object https://github.com/maakbaas/esp8266-iot-framework/blob/master/src/fetch.cpp#L11-L20 that could be acquired for use with other libraries. In my case the Adafruit MQTT library.

I'll give this a whirl tomorrow, but just wanted to leave an issue here so I don't forget about it (it's 3am :S)  
basically I'd move the https setup linked above to a separate public function, and then just call that from begin. That'd allow people to call that separately if they wanted, and just get the WiFiClientSecure object, to hand off to other libraries.

**Comment by Duckle29** [comment]
As I feared. That's not as simple as I had hoped.

I think there'd be a huge benefit if it was possible to break out the https component of this library into it's own library, and then have that provide a BearSSL::WiFiClientSecure *httpsClient useable with other libraries. Any pointers on how one would go about that? 

Seems to me like stuff such as the certstore generation and the certStore src files should be carried over, as well as the initialization code from fetch

**Comment by Duckle29** [comment]
To clarify, making the https object public, and using it in the [Adafruit MQTT Library](https://github.com/adafruit/Adafruit_MQTT_Library) caused run-time crashes, I think due to trying to access a null pointer.

```
--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (28):
epc1=0x4021cf92 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000

>>>stack>>>

ctx: cont
sp: 3ffffde0 end: 3fffffc0 offset: 0190
3fffff70:  00121100 3fffff90 3ffefb6c 40218310  
3fffff80:  3fffdad0 00000000 3ffefb6c 3ffeffc8
3fffff90:  3fffdad0 3ffefbbc 3ffefb28 40207fa9
3fffffa0:  feefeffe 00000000 3ffeff88 402196c0
3fffffb0:  feefeffe feefeffe 3ffe85e8 40100df9  
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------
```

![image](https://user-images.githubusercontent.com/2756925/105751624-91005680-5f46-11eb-9850-f728dec781bb.png)

**Comment by Duckle29** [comment]
Never mind. This functionality is available in the core secure wificlass now [certificate store](https://arduino-esp8266.readthedocs.io/en/latest/esp8266wifi/bearssl-client-secure-class.html#certificate-stores)
